### ISSUE TITLE
Limit repair's impact on the cluster, even when there are materialized views

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#4403]
We try to limit the impact of repair operation on the cluster by running the repair work, in both master and slave, in separate CPU and IO scheduling groups.

However, this isolation breaks down when the repaired table has materialized views. In this case, the repair process (on both master and slave nodes) saves new data in new sstables, and then generates view updates for these data, same as we do in view-building. These view updates are sent to other nodes (view replicas). Those nodes don't know that this work came from repair, so this work is not accounted to the repair CPU and IO groups. In the extreme case where there are many materialized views on the table, the cluster can become 100% occupied with this view-update work, even though the repair code is only using (say) 10% of the resources. Moreover, we don't even apply MV flow-control while view-building on the sstables received by repair, so the repair can generate growing and unlimited view backlogs.

To account for the view-update as part of repair, we need to use a separate verb, and separate connection, for the view updates generated from repair-generated sstables. But this is not enough, as the write-to-memtable on the view table is just a small part of the work - the more significant part (especially in IO) is the flush of these memtables to sstables, and this will not be accounted to the repair IO group. To do such accounting, we'll need to have separate memtables for view updates received via repair.

Finally, even doing all of the above, is not a replacement to enabling MV flow control on repair-view-building, which we'll need to do as well. The issue is that if the base-table repair process (receiving new base-table data) and the view updates share (say) 10% of each nodes resources, there is still no guarantee how much of this 10% each of the two part gets. If the base-table part gets too much of this 10%, it will generate view updates which will start to pile up in queues. So MV flow control will fix that (if the view backlog increases, we start slowing down the rate at which we build the views from sstables).

**Comment by duarten** [comment]
It's not true repair can generate unlimited view backlogs. Each shards only sends view updates for one view at any given time, synchronously (so there's natural backpressure). On the node being repaired we also throttle repair based on the number of pending sstables for which we need to generate view updates.

**Comment by nyh** [comment]
@duarten you are correct , thanks. I updated my issue description above, so your comment is no longer relevant - feel free to remove it if you think my description is now correct.

**Comment by duarten** [comment]
It's still not clear to me how the cluster can become 100% occupied.

**Comment by nyh** [comment]
I think Durarte's observation was correct - the view builder process only has limited concurrency and if it waits for the remote view updates to be completed before preparing new ones, endless amounts of view updates **cannot** accumulate. However, this is still extra work that is not accounted to the repair ("maintenance") scheduling group, which isn't good. We don't really have control whether this can increase repair from 10% CPU usage to 11% or to 50%. However, I never actually measured this effect - it might be very small which would make this issue not worth fixing.
